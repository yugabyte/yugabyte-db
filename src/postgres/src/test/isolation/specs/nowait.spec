# Test NOWAIT when regular row locks can't be acquired.

setup
{
  CREATE TABLE foo (
	id int PRIMARY KEY,
	data text NOT NULL
  );
  INSERT INTO foo VALUES (1, 'x');
}

teardown
{
  DROP TABLE foo;
}

session s1
setup		{ BEGIN; }
step s1a	{ SELECT * FROM foo FOR UPDATE NOWAIT; }
step s1b	{ COMMIT; }

session s2
setup		{ BEGIN; }
step s2a	{ SELECT * FROM foo FOR UPDATE NOWAIT; }
step s2b	{ COMMIT; }
