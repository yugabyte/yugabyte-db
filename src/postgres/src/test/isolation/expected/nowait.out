Parsed test spec with 2 sessions

starting permutation: s1a s1b s2a s2b
step s1a: SELECT * FROM foo FOR UPDATE NOWAIT;
id|data
--+----
 1|x   
(1 row)

step s1b: COMMIT;
step s2a: SELECT * FROM foo FOR UPDATE NOWAIT;
id|data
--+----
 1|x   
(1 row)

step s2b: COMMIT;

starting permutation: s1a s2a s1b s2b
step s1a: SELECT * FROM foo FOR UPDATE NOWAIT;
id|data
--+----
 1|x   
(1 row)

step s2a: SELECT * FROM foo FOR UPDATE NOWAIT;
ERROR:  could not obtain lock on row in relation "foo"
step s1b: COMMIT;
step s2b: COMMIT;

starting permutation: s1a s2a s2b s1b
step s1a: SELECT * FROM foo FOR UPDATE NOWAIT;
id|data
--+----
 1|x   
(1 row)

step s2a: SELECT * FROM foo FOR UPDATE NOWAIT;
ERROR:  could not obtain lock on row in relation "foo"
step s2b: COMMIT;
step s1b: COMMIT;

starting permutation: s2a s1a s1b s2b
step s2a: SELECT * FROM foo FOR UPDATE NOWAIT;
id|data
--+----
 1|x   
(1 row)

step s1a: SELECT * FROM foo FOR UPDATE NOWAIT;
ERROR:  could not obtain lock on row in relation "foo"
step s1b: COMMIT;
step s2b: COMMIT;

starting permutation: s2a s1a s2b s1b
step s2a: SELECT * FROM foo FOR UPDATE NOWAIT;
id|data
--+----
 1|x   
(1 row)

step s1a: SELECT * FROM foo FOR UPDATE NOWAIT;
ERROR:  could not obtain lock on row in relation "foo"
step s2b: COMMIT;
step s1b: COMMIT;

starting permutation: s2a s2b s1a s1b
step s2a: SELECT * FROM foo FOR UPDATE NOWAIT;
id|data
--+----
 1|x   
(1 row)

step s2b: COMMIT;
step s1a: SELECT * FROM foo FOR UPDATE NOWAIT;
id|data
--+----
 1|x   
(1 row)

step s1b: COMMIT;
